tests/data_association/test_point3d_initializer.py

```diff
@@ -218,3 +218,56 @@ def testSimpleTriangulationOnDoorDataset(self):
             if triangulated_track is None:
                 # assert we have failures which are already expected
                 self.assertIn(track_2d, expected_failures)
+
+
+class TestPoint3dInitializerUnestimatedCameras(unittest.TestCase):
+    """Unit tests for Point3dInitializer when a camera pose could not be estimated."""
+
+    def setUp(self):
+        """Data taken from Skydio-CraneMast-8, with images resized to 760 px resolution."""
+        super().setUp()
+
+        fx, k1, k2, u0, v0 = 583.1175, 0, 0, 507, 380
+        calibration = Cal3Bundler(fx, k1, k2, u0, v0)
+
+        wRi1 = np.array(
+            [
+                [-0.736357028, -0.589757459, 0.331608908],
+                [0.565525823, -0.805544778, -0.176856308],
+                [0.371428151, 0.0573040157, 0.926691631],
+            ]
+        )
+        wti1 = np.array([-0.649658109, 0.656963354, -0.382548681])
+        wTi1 = Pose3(Rot3(wRi1), wti1)
+
+        wRi2 = np.array(
+            [
+                [-0.802732442, -0.594660358, 0.0447178336],
+                [0.59626145, -0.80157983, 0.0440688035],
+                [0.009638943, 0.0620389785, 0.998027182],
+            ]
+        )
+        wti2 = np.array([-1.95517763, 1.44100216, -0.442089696])
+        wTi2 = Pose3(Rot3(wRi2), wti2)
+
+        # should be 3 cameras, but one is unestimated
+        cameras = {1: PinholeCameraCal3Bundler(wTi1, calibration), 2: PinholeCameraCal3Bundler(wTi2, calibration)}
+
+        self.triangulator = Point3dInitializer(cameras, TriangulationParam.NO_RANSAC, reproj_error_thresh=5)
+
+    def test_extract_measurements_unestimated_camera(self) -> None:
+        """Ensure triangulation args are None for length-2 tracks where one or more measurements come from
+        unestimated cameras.
+
+        Since camera 0 is unestimated, we have only 1 valid view within the track, so triangulation cannot succeed later.
```

Comment:
I think this would make more sense as an in-line comment near the zero ID

Reply:
good point, should be updated now